### PR TITLE
First Interfaces Iteration

### LINKED ISSUE #1: Discuss and Review Interfaces

### PR DESCRIPTION
**Pull request by FrancescoVaiani**
Add interfaces for minimum hardware infos
Fixes #1 

Signed-off-by: Francesco Vaiani <francesco.vaiani@secomind.com>

### CODE REVIEW

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    {
+      "endpoint": "/mem/total",
+      "type": "integer",
+      "description": "Total RAM quantity (MB)"
+    },
+    {
+      "endpoint": "/storage/total",
+      "type": "integer",
+      "description": "Total storage quantity (MB)"
+    }
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:38]: let's remove this, it gets tricky when there is more than one store unit, so I would postpone it to a dedicated interface.

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  ]
+}
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:40]: Missing new line at the end of file

File: io.edgehog.devicemanager.MobileConnectionInfo.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      "type": "string",
+      "description": "Mobile Network Modem Identifier"
+    },
+    {
+      "endpoint": "/imsi",
+      "type": "string",
+      "description": "SIM Identifier code"
+    }
+  ]
+}
```
Comment (bettio) [inline io.edgehog.devicemanager.MobileConnectionInfo.json:25]: Missing new line at the end of file

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+  "interface_name": "io.edgehog.devicemanager.HardwareInfo",
+  "version_major": 1,
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:3]: let's change major to 0

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  "version_minor": 0,
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:4]: let's change minor to 1

File: io.edgehog.devicemanager.HardwareInfo.json (hunk 2 above)
Comment (Pavinati) [inline io.edgehog.devicemanager.HardwareInfo.json:40]: missing newline character

File: io.edgehog.devicemanager.MobileConnectionInfo.json (hunk 3 above)
Comment (Pavinati) [inline io.edgehog.devicemanager.MobileConnectionInfo.json:25]: missing newline character

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      "type": "string",
+      "description": "CPU Model Display Name"
+    },
+    {
+      "endpoint": "/cpu/vendor",
+      "type": "string",
+      "description": "CPU Vendor"
+    },
+    {
+      "endpoint": "/mem/total",
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:30]: totalBytes

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      "type": "integer",
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:31]: longinteger

File: io.edgehog.devicemanager.HardwareInfo.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      "description": "Total RAM quantity (MB)"
```
Comment (bettio) [inline io.edgehog.devicemanager.HardwareInfo.json:32]: let's use bytes

File: io.edgehog.devicemanager.MobileConnectionInfo.json [hunk 9]
```diff
@@ -0,0 +1,25 @@
+{
+  "interface_name": "io.edgehog.devicemanager.MobileConnectionInfo",
+  "version_major": 1,
+  "version_minor": 0,
+  "type": "properties",
+  "ownership": "device",
+  "description": "General hardware capabilities",
+  "mappings": [
+    {
+      "endpoint": "/apn",
```
Comment (bettio) [inline io.edgehog.devicemanager.MobileConnectionInfo.json:10]: maybe we should make it parametric, when dealing with more than one modem. Anyway I'm not sure if we are already ready to add modem info now.
@rbino: any thought?

**Review by bettio — CHANGES_REQUESTED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**